### DISCUSSION TITLE
`@alwatr/element`

### DISCUSSION THREAD

**Discussion by MM25Zamanian** [AliMD/alwatr#282] [Ideas]
# Alwatr element option

- [x] `@alwatr/logger`
- [x] `@alwatr/signal`
- [x] `@alwatr/i18n`
- [ ] base element for pages (SEO friendly options) 

## `@alwatr/logger`

- [x] `constructor`
- [x] `connectedCallback`
- [x] `disconnectedCallback`
- [x] `firstUpdated`
- [x] `dispatchEvent`
- [x] `requestUpdate` 

## `@alwatr/signal`

- [x] signal interface shorter
- [x] automatically listener remover
- [ ] log signal operations (???) 

## `@alwatr/i18n`

After rewriting `i18n`, I will write its tasks

## base element for pages (SEO friendly options)

[`IBM:pwa-lit-template`](https://github.com/IBM/pwa-lit-template/blob/main/src/helpers)

- [ ] meta tags by page
- [ ] title by page (I think this feature should in router)

**Comment by MM25Zamanian** [comment]
@AliMD This discussion was started on the way from Najaf to Karbala

and finished in Karbala

**Comment by MM25Zamanian** [comment]
What do you think ?

@asmanp2012 @njfamirm

**Comment by njfamirm** [comment]
لاگر داره

**Reply by MM25Zamanian** [reply]
دقت کن
از لاگر فقط به تسک مونده

لطفا کامل بخون

**Comment by alimd** [comment]
چه عالی
منو اینو الان دیدم
کی انجامش بدیم؟
